### ISSUE TITLE
New column type: NumericColumn

### ISSUE THREAD

**Issue by Seb33300** [stwe/DatatablesBundle#577]
Hi,

Do you plan to add other column types like a `NumericColumn`?

This column could have options to:
- set the round precision with round type (floor, ceil, ...)
- set a fixed number of zero to display numbers like '12.30' instead of '12.3'
- set a prefix / suffix (to simply add units, % or money symbols...)
- other ideas ?

I know this can be done with virtual columns, but this could be very helpful to have this instead of creating a virtual column every time.

Seb

**Comment by stwe** [comment]
I have discarded new features for reasons of time.

Currently, I am working on the ImageColumn. The idea with the NumericColumn I had times before. With the variable "type_of_field" I also wanted to create Columns "on the fly". At some point I have also holiday. Then I have more time.
Thank you for your ideas and your commitment.

Are you planning to work on this bundle any longer or is your cooperation only project-related?

**Comment by Seb33300** [comment]
I am working on an application for a customer which have many datatables.

So I am contributing when features I really need to achieve this app are missing.

My contract with my customer will still continue to the end of the year, so I will may have more contributions to commit.
But I have other priorities at the moment.

**Comment by stwe** [comment]
see: https://github.com/stwe/DatatablesBundle/commit/3cc35ee9da21f8839c33cdfed9c0cdc71bbb0586

But there is a problem with the in-place edit and the input of "." and ","

**Comment by Seb33300** [comment]
Hello!

Thanks for that!

Only one question: 

Why do you provide `use_format_currency` and `currency` options?

Since it needs to pass a `NumberFormatter` object, the developer can pass it already defined with the correct currency format he wish. 
(We can just provide some examples (currency, percent, etc...) of `NumberFormatter` to pass to the column in the documentation)
If no formatter is passed, just create a default `NumberFormatter `formatting a number based on the user locale.

**Comment by stwe** [comment]
> Why do you provide use_format_currency and currency options?

The NumberFormatter has two different methods to format the input:

```
Public string formatCurrency (float $ value, string $ currency)
Public string format (number $ value [, int $ type])
```

The `use_format_currency` option calls the` formatCurrency` method instead of `format`.

**Comment by Seb33300** [comment]
Ok. Maybe you could only provide the `currency` option. and call `formatCurrency` if this option is filled?

Otherwise, I think the formatter option should be optional (NumberFormatter::DECIMAL by default).

**Comment by stwe** [comment]
> Ok. Maybe you could only provide the currency option. and call formatCurrency if this option is filled?

👍 

> Otherwise, I think the formatter option should be optional 

Or formatter is required and I only check if an object was passed (check without if instanceof \NumberFormatter). So anyone can create their own formatter object, with custom methods `->format` and `->formatCurrency`.

**Comment by stephanvierkant** [comment]
This issue can be closed, isn't it? See 3cc35ee9da21f8839c33cdfed9c0cdc71bbb0586
